### ISSUE TITLE
Support for emails with body charset Windows-31J 

### ISSUE THREAD

**Issue by ramirser000** [roundcube/roundcubemail#8869]
It looks like if the email body has a charset like below, it will result in an invalid charset result when the function parse_charset() evaluates it
```
Content-Type: text/plain; charset=Windows-31J
Content-Transfer-Encoding: base64
```
It results in WINDOWS31 which is invalid.

**Comment by alecpl** [comment]
We should make it an alias of `CP932`  (works with both iconv and mbstring). It could also be `SJIS-WIN`, I guess, but I'm not sure whether that would be better or not (https://github.com/php/php-src/issues/8308, https://en.wikipedia.org/wiki/Code_page_932_(Microsoft_Windows)).

**Comment by alecpl** [comment]
Fixed.

**Comment by ramirser000** [comment]
Hello,

By any chance can this fix also be applied to the 1.4 LTS version?
